tiledb/sm/filesystem/vfs.cc

```diff
@@ -856,7 +856,9 @@ Status VFS::read_impl(
 
 Status VFS::read_all(
     const URI& uri,
-    const std::vector<std::tuple<uint64_t, void*, uint64_t>>& regions) {
+    const std::vector<std::tuple<uint64_t, void*, uint64_t>>& regions,
+    ThreadPool* thread_pool,
```

Comment:
Why not use the 'thread_pool_' within this class?

Reply:
We will revisit this in a subsequent PR.